src/averageAges.js

```diff
@@ -15,11 +15,12 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people
+    .filter(person => person.sex === 'm')
+    .filter(person => !century || century === Math.ceil(person.died / 100))
+    .map(person => person.died - person.born);
+
+  return men.reduce((a, b) => a + b) / men.length;
```

Comment:
Checklist p2, correct in whole work
```suggestion
  return men.reduce((a, b) => (a + b)) / men.length;
```